fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashActivity.java

```diff
@@ -19,212 +19,209 @@
 
 import dagger.hilt.android.AndroidEntryPoint;
 
-/**
- * Activity for the digital cash
- */
+/** Activity for the digital cash */
 @AndroidEntryPoint
 public class DigitalCashActivity extends AppCompatActivity {
-    private DigitalCashViewModel mViewModel;
-    public static final String TAG = DigitalCashActivity.class.getSimpleName();
-
-    @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.digital_cash_main_activity);
-        mViewModel = obtainViewModel(this);
-        setupNavigationBar();
-
-
-        setTheIntent();
-        // Subscribe to "open home"
-        openHomeEvent();
-        // Subscribe to "open history"
-        openHistoryEvent();
-        // Subscribe to "open send"
-        openSendEvent();
-        // Subscribe to "open receive"
-        openReceiveEvent();
-        // Subscribe to "open issue"
-        openIssueEvent();
-        // Subscribe to "open receipt"
-        openReceiptEvent();
-        mViewModel.openHome();
+  private DigitalCashViewModel mViewModel;
+  public static final String TAG = DigitalCashActivity.class.getSimpleName();
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.digital_cash_main_activity);
+    mViewModel = obtainViewModel(this);
+    setupNavigationBar();
+
+    setTheIntent();
+    // Subscribe to "open home"
+    openHomeEvent();
+    // Subscribe to "open history"
+    openHistoryEvent();
+    // Subscribe to "open send"
+    openSendEvent();
+    // Subscribe to "open receive"
+    openReceiveEvent();
+    // Subscribe to "open issue"
+    openIssueEvent();
+    // Subscribe to "open receipt"
+    openReceiptEvent();
+    mViewModel.openHome();
+  }
+
+  public void setTheIntent() {
+    if (getIntent().getExtras() != null) {
+      String id = getIntent().getExtras().getString(Constants.LAO_ID_EXTRA, "");
+      mViewModel.subscribeToLao(id);
+      mViewModel.setLaoId(getIntent().getExtras().getString(Constants.LAO_ID_EXTRA, ""));
```

Comment:
```suggestion
      mViewModel.setLaoId(id);
```
In term of perf isn't it better to use that one ?

Reply:
True